remote/bitbucketserver/internal/client.go

```diff
@@ -214,3 +250,142 @@ func (c *Client) paginatedRepos(start int) ([]*Repo, error) {
 	}
 	return repoResponse.Values, nil
 }
+
+func Filter(vs []string, f func(string) bool) []string {
+	vsf := make([]string, 0)
+	for _, v := range vs {
+		if f(v) {
+			vsf = append(vsf, v)
+		}
+	}
+	return vsf
+}
+
+//TODO: find a clean way of doing these next two methods- bitbucket server hooks only support 20 cb hooks
+func arrayToHookSettings(hooks []string) HookSettings {
```

Comment:
It's part of the api for updating the hooks.  I wish it was better :/ I imagine there's a way to not have to list out every struct property in goLang and make it a bit less verbose(probably using introspection), I just haven't had a chance to look at it yet.